### DISCUSSION TITLE
Is there a way to control job queue time on self-hosted runners aka expire jobs if not running in X minutes

### DISCUSSION THREAD

**Discussion by atodorov** [community/community#160687] [Actions]
Hey folks,
I am using self-hosted runners quite a lot where my runners are provisioned via a matrix job just before they are needed. Do teting on the actual self-hosted runner & then clean up. 

Recently I've had the following situation:

1) My provisioning job was stuck and kept retrying for 6 hours before it timed out
2) The real job which was waiting for this runner was scheduled and stayed in the queue for 24hrs before GitHub considered it to be expired and terminated it. From the [docs](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners#usage-limits)

> Job queue time - Each job for self-hosted runners that has been queued for at least 24 hours will be canceled. The actual time in queue can reach up to 48 hours before cancellation occurs. If a self-hosted runner does not start executing the job within this limit, the job is terminated and fails to complete.

This lead to issues for my clean-up b/c it was waiting for 2) to complete. 

I can solve 1) by using [jobs.<job_id>.timeout-minutes](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes). 

I am looking for a similar solution for item 2). Is there any way to control how long before a job in the queue is considered stale and should be terminated ? I want to bring this number way down from the default of 24 hrs.

**Comment by yaobinwen-mvs** [comment]
Google took me here. I wonder if you will have to use some GitHub API to query the jobs that have been in the queue for a too long time and cancel them by calling another API. I know a scheduled workflow execution can be canceled on the GitHub web UI so I assume there should be an API for the cancellation too.
